internal/processing/processors/authorization_policy_processor.go

```diff
@@ -34,34 +40,101 @@ func (r AuthorizationPolicyProcessor) EvaluateReconciliation(ctx context.Context
 	return changes, nil
 }
 
-func (r AuthorizationPolicyProcessor) getDesiredState(api *gatewayv1beta1.APIRule) []*securityv1beta1.AuthorizationPolicy {
-	return r.Creator.Create(api)
+func (r AuthorizationPolicyProcessor) getDesiredState(api *gatewayv1beta1.APIRule) (map[string][]*securityv1beta1.AuthorizationPolicy, error) {
+	aps, err := r.Creator.Create(api)
+	if err != nil {
+		return nil, err
+	}
+	return aps, nil
 }
 
-func (r AuthorizationPolicyProcessor) getActualState(ctx context.Context, client ctrlclient.Client, api *gatewayv1beta1.APIRule) ([]*securityv1beta1.AuthorizationPolicy, error) {
+func (r AuthorizationPolicyProcessor) getActualState(ctx context.Context, client ctrlclient.Client, api *gatewayv1beta1.APIRule) (map[string]map[string]*securityv1beta1.AuthorizationPolicy, error) {
 	labels := processing.GetOwnerLabels(api)
 
 	var apList securityv1beta1.AuthorizationPolicyList
 	if err := client.List(ctx, &apList, ctrlclient.MatchingLabels(labels)); err != nil {
 		return nil, err
 	}
 
-	authorizationPolicies := append([]*securityv1beta1.AuthorizationPolicy{}, apList.Items...)
+	authorizationPolicies := make(map[string]map[string]*securityv1beta1.AuthorizationPolicy)
+	for i, ap := range apList.Items {
+		index, ok := ap.Labels[processing.IndexLabelName]
+		if !ok {
```

Comment:
Isn't this more or less a migration case? I think we can ignore something like this and just re-create APs in this case for now as it should happen only once for currently existing APs.

Reply:
If we don't store the APs without labels in some way, they won't be deleted and this will result in AP duplication. Technically this should only be the case right now so I can delete this but I'm not 100% sure if it is OK to have them duplicated.

Reply:
That being said, I think that the `else if` statement at line 110 should be OK to delete, as this will only cause the APs to be recreated

Reply:
https://github.com/kyma-project/api-gateway/pull/225/commits/5fd017c1b88a4402982f9ee960922ce71176844f